### PR TITLE
[INTEL MKL] Upgrading RequantizePerChannel Op with API changes in MKLDNN 1.0

### PR DESCRIPTION
**Pull request by mdfaijul**
This PR ports RequantizePerChannel operator to MKLDNN 1.0

### CODE REVIEW

**Comment by rthadur** [comment]
@mdfaijul do you want to add test cases around new changes ?

File: tensorflow/core/kernels/mkl_requantize_per_channel_op.cc [hunk 1]
```diff
@@ -120,16 +120,28 @@ class MklRequantizePerChannelOp : public OpKernel {
...
+      reorder_stream.submit(
+          {mkldnn::reorder(reorder_pd, *input_mem_prim, *output_mem_prim)});
+#else
+      std::unordered_map<int, mkldnn::memory> reorder_args = {
+          {MKLDNN_ARG_FROM, *input_mem_prim},
+          {MKLDNN_ARG_TO, *output_mem_prim}};
+      std::unique_ptr<mkldnn::primitive> reorder_prim(
+          new mkldnn::reorder(reorder_pd));
+      reorder_prim->execute(reorder_stream, reorder_args);
+#endif  // ENABLE_MKLDNN_V1
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_requantize_per_channel_op.cc]: Nit, since it's `#ifndef`.
```suggestion
#endif  // !ENABLE_MKLDNN_V1
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I have one minor nit.

@rthadur I think it's fine. This PR doesn't add any new functionality. It's just a workaround to make the same code work with both versions of the MKL-DNN library (v0.x and v1.x) which have incompatible APIs.

**Comment by mdfaijul** [comment]
@penpornk Thanks! Have changed it. Please take a look.

**Review by penpornk — APPROVED** [review]
Thank you for the quick fix!

**Pull request merged**